pkg/github/actions.go

```diff
@@ -775,6 +787,16 @@ func downloadLogContent(logURL string) (string, *http.Response, error) {
 
 	// Clean up and format the log content for better readability
 	logContent := strings.TrimSpace(string(content))
+
+	// Truncate to tail_lines if specified
+	if tailLines > 0 {
+		lines := strings.Split(logContent, "\n")
+		if len(lines) > tailLines {
+			lines = lines[len(lines)-tailLines:]
+			logContent = strings.Join(lines, "\n")
+		}
+	}
```

Comment:
Nerdy, but you could build from reverse parsing the string maybe, for efficiency perhaps?

```
if tailLines > 0 {
    lineCount := 0
    lastNewlinePos := len(logContent)
    
    // Count backwards to find the nth newline from the end
    for i := len(logContent) - 1; i >= 0 && lineCount < tailLines; i-- {
        if logContent[i] == '\n' {
            lineCount++
            if lineCount == tailLines {
                logContent = logContent[i+1:]
                break
            }
        }
    }
}
```